src/libawkward/io/json.cpp

```diff
@@ -101,16 +101,28 @@ namespace awkward {
     rj::Writer<rj::StringBuffer> writer_;
   };
 
-  ToJsonString::ToJsonString(int64_t maxdecimals)
-      : impl_(new ToJsonString::Impl(maxdecimals)) { }
+  ToJsonString::ToJsonString(int64_t maxdecimals,
+    const char* nan_string,
+    const char* infinity_string,
+    const char* minus_infinity_string)
+      : impl_(new ToJsonString::Impl(maxdecimals))
+      , nan_string_(nan_string)
+      , infinity_string_(infinity_string)
+      , minus_infinity_string_(minus_infinity_string) {
+    }
 
   ToJsonString::~ToJsonString() {
     delete impl_;
   }
 
   void
   ToJsonString::null() {
-    impl_->null();
+    if (nan_string_ != nullptr) {
+      impl_->string(nan_string_, (int64_t)strlen(nan_string_));
+    }
+    else {
+      impl_->null();
+    }
```

Comment:
Awkward missing values ("None" in Python) is not NaN and should not be replaced by the `nan_string`, so this change should be reverted.

Reply:
done